tests/unit/h/security/policy/combined_test.py

```diff
@@ -1,150 +1,169 @@
-from unittest.mock import patch, sentinel
+from unittest.mock import create_autospec, sentinel
 
 import pytest
 
 from h.security.policy import SecurityPolicy
-
-# pylint: disable=protected-access
+from h.security.policy.combined import applicable_policies, call_policies
 
 
 class TestSecurityPolicy:
-    def test_construction(self, BearerTokenPolicy, AuthClientPolicy, CookiePolicy):
-        policy = SecurityPolicy(proxy_auth=False)
+    def test_remember(self, call_policies, policy, pyramid_request):
+        headers = policy.remember(pyramid_request, sentinel.userid, foo=sentinel.foo)
 
-        BearerTokenPolicy.assert_called_once_with()
-        assert policy._bearer_token_policy == BearerTokenPolicy.return_value
-        AuthClientPolicy.assert_called_once_with()
-        assert policy._http_basic_auth_policy == AuthClientPolicy.return_value
-        CookiePolicy.assert_called_once_with()
-        assert policy._ui_policy == CookiePolicy.return_value
+        call_policies.assert_called_once_with(
+            "remember", pyramid_request, sentinel.userid, foo=sentinel.foo
+        )
+        assert headers == call_policies.return_value
```

Comment:
Since `call_policies()` is now a unit of its own with its own unit tests (rather than a private helper method) the tests for the multiple higher-level methods that call `call_policies()` don't all need to re-test the behaviour of `call_policies()`. Instead they mock `call_policies()` and just test how the higher-level method calls it.

This avoids all the mess with single tests that tested the behaviour of the private `_call_sub_policies()` helper via all three higher-level public methods; or that patched the private helper; or that picked just one higher-level method to test as a stand-in for the others (which of course means you haven't actually tested the other methods). The previous tests used all three of these approaches in one test file!